### PR TITLE
UI module improvements

### PR DESCRIPTION
**Pull request by jechter**
- Redesigned `UIActionInputModule`
	 * Added a button in the inspector to automatically assign actions from an input action asset based on commonly used action names.
	 * Will now populate actions with useful defaults.
	 * Removed `clickSpeed` property - will use native click counts from the OS where available instead.
	 * Removed `sendEventsWhenInBackground` property.
	 * Hiding `Touches` and `TrackedDevices` until we decide how to handle them.
	 * Remove `moveDeadzone` property as it is made redundant by the action's dead zone.
	 * Removed `UIActionInputModuleEnabler` component, `UIActionInputModule` will now enable itself.
- Changed default button press point to 0.5.
- Fixed scrolling in `UIActionInputModule`.

Screen Shot 2019-04-26 at 11 23 18 AM

### CODE REVIEW

File: Assets/Tests/InputSystem/CoreTests_Actions.cs [hunk 1]
```diff
@@ -3404,6 +3404,9 @@ public void Actions_CanPerformContinuousHold()
     [Property("TimesliceEvents", "Off")]
     public void Actions_CanPerformHoldOnTrigger()
     {
+        var oldPressPoint = InputSystem.settings.defaultButtonPressPoint;
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/CoreTests_Actions.cs]: This is actually not necessary. A new settings object is created for every test and then thrown away.

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 2]
```diff
@@ -46,6 +46,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
 - Tests are now no longer part of the com.unity.inputsystem package. The `InputTestFixture` class still is for when you want to write input-related tests for your project. You can reference the `Unity.InputSystem.TestFixture` assembly when you need to do that.
+- Redesigned `UIActionInputModule`
+ * Added a button in the inspector to automatically assign actions from an input action asset based on commonly used action names.
+ * Will now populate actions with useful defaults.
+ * Removed `clickSpeed` property - will use native click counts from the OS where available instead.
+ * Removed `sendEventsWhenInBackground` property.
+ * Hiding `Touches` and `TrackedDevices` until we decide how to handle them.
+ * Remove `moveDeadzone` property as it is made redundant by the action's dead zone.
+ * Removed `UIActionInputModuleEnabler` component, `UIActionInputModule` will now enable itself.
+- Changed default button press point to 0.5.
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: The changelog entries ended up in the 0.2.8 section here. There's a new section for 0.3 (hopefully this time it'll be that) above.

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
+#if UNITY_EDITOR
+
+using UnityEditor;
+using UnityEngine.Experimental.Input.Editor;
+
+namespace UnityEngine.Experimental.Input.Plugins.UI.Editor
+{
+    [CustomEditor(typeof(UIActionInputModule))]
+    internal class UIActionInputModuleEditor: UnityEditor.Editor
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs]: Can we have one of these for `StandaloneInputModule` to have a one-click switch over to this one?

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                            if (reference.m_ActionId == action.m_Id)
+                                return new InputActionProperty(reference);
+                        }
+                    }
+                }
+            }
+            return defaultValue;
+        }
+
+        private enum ActionReferenceType {
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs]: Please run format.sh before committing.

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            Reference,
+            SerializedData
+        };
+
+        private string[] m_ActionNames = new[]
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs]: static?

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+            }
+
+            GUILayout.BeginVertical(EditorStyles.helpBox);
+            EditorGUI.indentLevel++;
+            m_ActionsFoldout = EditorGUILayout.Foldout(m_ActionsFoldout, "Actions", Styles.s_FoldoutStyle);
+            EditorGUI.indentLevel--;
+
+            if (m_ActionsFoldout)
+            {
+                const string buttonLabel = "Read Actions from Asset…";
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModuleEditor.cs]: "Link Actions from Asset..."?

File: Packages/com.unity.inputsystem/InputSystem/Plugins/Users/InputUser.cs [hunk 7]
```diff
@@ -1613,7 +1613,10 @@ internal static void OnDeviceChange(InputDevice device, InputDeviceChange change
                     // while concurrently comparing whatever bits make it through to the default
                     // state buffer.
                     if (device.CheckStateIsAtDefaultIgnoringNoise())
+                    {
+                        Profiler.EndSample();
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/Users/InputUser.cs:1617]: Good catch!

**Review by Rene-Damm — CHANGES_REQUESTED** [review]
Looks good to me. Couple small things.

**Review by douglas-piconi — APPROVED** [review]

**Review by Rene-Damm — APPROVED** [review]

**Pull request merged**